float8_experimental/float8_aten_api.py

```diff
@@ -43,8 +63,18 @@ def mm_float8_emulated(
 lib.impl("mm_float8_emulated", mm_float8_emulated, "CPU")
 lib.impl("mm_float8_emulated", mm_float8_emulated, "CUDA")
 
+lib.define(
+    "cast_to_float8_tensor(Tensor x, Tensor x_scale, ScalarType float8_dtype, bool emulate) -> Tensor"
+)
+lib.impl("cast_to_float8_tensor", cast_to_float8_tensor, "CPU")
+lib.impl("cast_to_float8_tensor", cast_to_float8_tensor, "CUDA")
+
 
 @torch.library.impl(lib, "mm_float8_emulated", "Meta")
 def _mm_float8_emulated_meta(m1, s1, m2, s2, dtype3):
     out = torch.mm(m1.float(), m2.float()).to(dtype3)
     return out, torch.empty(1, device="meta")
+
+@torch.library.impl(lib, "cast_to_float8_tensor", "Meta")
+def _cast_to_float8_tensor_meta(x, x_scale, float8_dtype, emulate):
```

Comment:
@bdhirsh is this the correct way to register a pt2 compliant op? I think I might miss sth as the compile tests are failing a lot

Reply:
This might require some more work actually. This op probably will not do what you want as written.

(1) the impl here will need to return an actual `Float8Tensor()` subclass, since it is what runs during tracing

(2) We also need DTensor to handle this op in its torch_dispatch, to wrap the result in a DTensor

Reply:
We actually also might need to register this logic to FakeTensor directly, since it's pretty special. We want FakeTensor to handle this logic carefully, since we don't want it to (by default) try to convert the output directly into a FakeTensor.

Reply:
Is there a way to register this logic out of tree?